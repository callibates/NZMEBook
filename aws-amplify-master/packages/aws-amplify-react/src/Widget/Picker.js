/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';

import { I18n, Logger } from 'aws-amplify';
import AmplifyTheme from '../AmplifyTheme';

const PickerPicker = {
    position: 'relative'
}

const PickerPreview = {
    maxWidth: '100%'
}

const PickerButton = {
    width: '10em',
    height: '3em',
    fontSize: '1.2em',
    textAlign: 'center'
}

const PickerInput = {
    width: '100%',
    height: '100%',
    display: 'inline-block',
    position: 'absolute',
    left: 0,
    top: 0,
    opacity: 0,
    cursor: 'pointer'
}

const logger = new Logger('Picker');

export default class Picker extends Component {
    handleInput(e) {
        var that = this;

        const file = e.target.files[0];
        if (!file) { return; }

        const { name, size, type } = file;
        logger.debug(file);

        const { onPick } = this.props;
        if (onPick) {
            onPick({
                file: file,
                name: name,
                size: size,
                type: type
            });
        }
    }

    render() {
        const title = this.props.title || 'Pick a File';
        const accept = this.props.accept || '*/*';

        const theme = this.props.theme || AmplifyTheme;
        const pickerStyle = Object.assign(
            {},
            PickerPicker,
            theme.pickerPicker
        );
        const buttonStyle = Object.assign({}, PickerButton, theme.button, theme.pickerButton);
        const inputStyle = Object.assign({}, PickerInput, theme.pickerInput);

        return (
            <div style={pickerStyle}>
                <button style={buttonStyle}>
                    {I18n.get(title)}
                </button>
                <input
                    title={I18n.get(title)}
                    type="file" accept={accept}
                    style={inputStyle}
                    onChange={(e) => this.handleInput(e)}
                />
            </div>
        )
    }
}
