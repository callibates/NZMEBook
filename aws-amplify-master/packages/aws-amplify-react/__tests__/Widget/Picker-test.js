import React, { Component } from 'react';
import Picker from '../../src/Widget/Picker';

describe('Picker test', () => {
    describe('render test', () => {
        test('render correcly', () => {
            const wrapper = shallow(<Picker/>);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('interaction test', () => {
        test('chnage on input', () => {
            const spyon = jest.spyOn(Picker.prototype, 'handleInput').mockImplementationOnce(() => {
                return;
            });

            const wrapper = shallow(<Picker/>);

            wrapper.find('input').at(0).simulate('change', 'event');

            expect(spyon).toBeCalledWith('event');

            spyon.mockClear();
        });
    });

    describe('handleInput test', () => {
        test('happy case', () => {
            const onPickFn = jest.fn();

            const wrapper = shallow(<Picker/>);
            const picker = wrapper.instance();
            wrapper.setProps({
                onPick: onPickFn
            });

            const event = {
                target: {
                    files:[{
                        name: 'name',
                        size: 'size',
                        type: 'type'
                    }]
                }
            }
            picker.handleInput(event);
            
            expect(onPickFn).toBeCalledWith({"file": {"name": "name", "size": "size", "type": "type"}, 
                                            "name": "name", 
                                            "size": "size", 
                                            "type": "type"});
        });

        test('no onPick', () => {
            const wrapper = shallow(<Picker/>);
            const picker = wrapper.instance();

            const event = {
                target: {
                    files:[{
                        name: 'name',
                        size: 'size',
                        type: 'type'
                    }]
                }
            }
            picker.handleInput(event);
        });
    });


});