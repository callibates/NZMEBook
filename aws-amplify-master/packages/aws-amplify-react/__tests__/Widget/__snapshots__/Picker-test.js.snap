// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker test render test render correcly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <button
    style={
      Object {
        "backgroundColor": "#fff",
        "backgroundImage": "none",
        "border": "1px solid transparent",
        "borderColor": "#ccc",
        "borderRadius": "4px",
        "color": "#333",
        "cursor": "pointer",
        "display": "inline-block",
        "fontSize": "14px",
        "fontWeight": 400,
        "height": "3em",
        "lineHeight": "1.42857143",
        "marginBottom": "0",
        "padding": "6px 12px",
        "textAlign": "center",
        "touchAction": "manipulation",
        "userSelect": "none",
        "verticalAlign": "middle",
        "whiteSpace": "nowrap",
        "width": "10em",
      }
    }
  >
    Pick a File
  </button>
  <input
    accept="*/*"
    onChange={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "display": "inline-block",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    title="Pick a File"
    type="file"
  />
</div>
`;
